addons/mail/models/mail_activity_type.py

```diff
@@ -125,3 +125,50 @@ def _inverse_triggered_next_type_id(self):
                 activity_type.chaining_type = 'trigger'
             else:
                 activity_type.chaining_type = 'suggest'
+
+    def write(self, values):
+        """ Protect some master types against deletiong or modification when they
+        are used as default in apps, in critical flows, plans, ... """
+        if values.get('res_model'):
```

Comment:
I think there's a world where I write `res_model_id` instead of `res_model`, and `res_model` is updated without ever matching this branch. Shouldn't we check after the write to be sure? So any compute will have run.

Reply:
Don't see res_model_id on activity type model ?

Reply:
Activities have res_model_id, not type. Not sure why :shrug:

Reply:
Yes I think I was looking at the fields of `mail.activity` sorry, seems good then